### PR TITLE
Run 3 DCA extraction

### PR DESCRIPTION
**Pull request by victor-gonzalez**
I created it in a specific branch to have my master available

I will transfer the comments from the previous PR draft
After that I will update it with the comments implemented

### CODE REVIEW

File: Common/TableProducer/trackextension.cxx [hunk 1]
```diff
@@ -32,21 +36,44 @@ using namespace o2::framework::expressions;
...
           auto const& collision = track.collision();
-          trackPar.propagateParamToDCA({collision.posX(), collision.posY(), collision.posZ()}, magField, &dca);
+          if (cfgDcaMethod == 1) {
+            trackPar.propagateParamToDCA({collision.posX(), collision.posY(), collision.posZ()}, o2::base::Propagator::Instance()->getNominalBz(), &dca);
+          } else if (cfgDcaMethod == 2) {
+            o2::dataformats::DCA dcaInfo;
+            if (o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackPar, 2., matCorr, &dca)) {
+              dca[0] = dcaInfo.getY();
+              dca[1] = dcaInfo.getZ();
+            }
```
Comment (victor-gonzalez) [inline Common/TableProducer/trackextension.cxx:72]: From Ruben

you are passing to `propagateToDCABxByBz` the `dca` then overwrite it with `dcaInfo` which was never filled.

**Comment by victor-gonzalez** [comment]
@jgrosseo @shahor02 For converted Run 2 nothing of this is applicable? Or the proper 'calibration' files have to loaded?

**Comment by jgrosseo** [comment]
Yes, I think we should also do a run selection here. Ideally with different process functions as in the event selection

**Comment by victor-gonzalez** [comment]
Note that 

**PbPb**
- LHC15o **train 203**
  - `++` 8 runs
  - `--` 2 runs
  - one annoying run (245064)
- LHC20f6 **train 207**
  - `--` 50 runs
  - `++` 88 runs
  - three annoying runs (245064, 245146, 245345)
- LHC18q
  - `++` all runs
- LHC18r
  - `--` all runs
- LHC20g2a_2 **train 219**
  - `++` LHC18q runs
  - `--` LHC18r runs
- LHC20e3a **train 210**
  - `++` LHC18q runs
  - `--` LHC18r runs

**pp**
- LHC15n
  - `++` all runs
- LHC17p **train 212**
  - `--` all runs
- LHC17q **train 213**
  - `--` all runs
- LHC20a7_cent **train 209**
  - `--` all runs
- LHC18b LHC18g4 **train 208**
  - `--` all runs
- LHC18d
  - `--` all runs
- LHC18e
  - `--` all runs
- LHC18f
  - `--` all runs
- LHC18g
  - `++` all runs
- LHC18h
  - `--` all runs
- LHC18i
  - `++` all runs
- LHC18k
  - `-0` all runs
- LHC18l
  - `--` all runs
- LHC18m
  - `++` all runs
- LHC18n
  - `++` all runs
- LHC18o
  - `++` all runs but
  - one annoying run (293507)
- LHC18p
  - `++` all runs
- LHC18_pp_bdefghiklmnop **train 222**
  - according to each period

The current `trackextension` task assumes that the magnetic field is `5.0` for all analyses

But actually, I have not seen any difference up to the tenth decimal digit between the `dca` calculated with the current `trackextension` task (which considers positive magnetic field polarity) and with this PR incorporated, for the run `246391` (LHC15o) for which the magnetic field polarity is configured as negative according to its `GRP` object.

@jgrosseo  Does this make sense?

(the annoying runs are runs which in the RCT show a different magnetic field polarity of what appears in their `GRP` objects)

**Comment by jgrosseo** [comment]
I believe that for Run 2 data, the tracks are already propagated to the closest point to the vertex. Therefore, it does not matter what is the magnetic field setting. The function just calculates the DCA.

If you want to see a difference of magnetic field setting you can check my correlation task and what is done in `PairCuts.h`. Note that the magnetic field is still hard coded there. 
https://github.com/jgrosseo/O2Physics/blob/master/PWGCF/Core/PairCuts.h#L112

**Comment by victor-gonzalez** [comment]
Good point!
Thanks!

**Comment by jgrosseo** [comment]
I have tested the GRP object also in my code and it works fine: https://github.com/AliceO2Group/O2Physics/pull/245

Shall we go ahead and move this to review, @victor-gonzalez ?

**Comment by victor-gonzalez** [comment]
I like the way you used to extract the value of the magnetic field
I will update the PR and then we move it to review
While in review I will learn how to extract the information from the Logbook and fix the GRPs of the annoying runs

**Comment by victor-gonzalez** [comment]
@shahor02 @jgrosseo has anything changed lately regarding the material LUT?
Since a few days we are getting these messages when retrieving the material LUT from CCDB
```
[12418:track-extension]: Info in <TJAlienFile::Open>: Accessing file /alice/data/CCDB/GLO/Param/MatLUT/03/10766/a40de9d0-2c6f-11ec-ab62-2a010e0a09fb in SE <ALICE::CERN::EOS>
[12418:track-extension]: [10:03:04][WARN] Did not find object under expected name ccdb_object
[12416:timestamp-task]: [10:03:04][INFO] Is alien token present?: 1
[12418:track-extension]: [10:03:04][WARN] Did not find object under expected name ccdb_object
[12418:track-extension]: [10:03:04][ERROR] Couldn't retrieve object corresponding to o2::base::MatLayerCylSet from TFile
```

**Comment by victor-gonzalez** [comment]
For Run2 everything is ready
For Run3 probably not having the material LUT will affect the track propagation and impact the precision of the DCA extraction

**Comment by shahor02** [comment]
Hi @victor-gonzalez 

I see that the object in the file is written under the old `MatBud` name, should have been `ccdb_object`. Will upload correct one.

**Comment by victor-gonzalez** [comment]
@shahor02 Thanks for the prompt fix!!!

**Comment by shahor02** [comment]
@victor-gonzalez : done, you can access it both from the `alice-ccdb.cern.ch` and from `http://ccdb-test.cern.ch:8080` as 
```
auto& ccdbMgr = o2::ccdb::BasicCCDBManager::instance();
ccdbMgr.setURL("https://alice-ccdb.cern.ch");
auto mb = o2::base::MatLayerCylSet::rectifyPtrFromFile(ccdbMgr.get<o2::base::MatLayerCylSet>("GLO/Param/MatLUT"));
```

**Comment by victor-gonzalez** [comment]
@shahor02 I see the new object in CCDB but still with the same filename `matbud.root`. Is that correct?

**Comment by victor-gonzalez** [comment]
@shahor02 Everything correct now!!
Tests passed correctly!
Thanks!

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**